sdk/src/asset_handlers/jpeg_io.rs

```diff
@@ -1129,6 +1172,45 @@ pub mod tests {
         assert!(read_xmp.contains(test_msg));
     }
 
+    #[test]
+    fn test_xmp_read_write_stream() {
+        let source = crate::utils::test::fixture_path("CA.jpg");
+
+        let temp_dir = tempfile::tempdir().unwrap();
+        let output = crate::utils::test::temp_dir_path(&temp_dir, "CA_test.jpg");
```

Comment:
Should we considering adding test buffers or test streams to our test utils in the future? It would be nice if we can have this running in environments that don't support file I/O, such as part of the test-wasm suite.

Reply:
Modified to pure streams and testable in WASM now

Reply:
Awesome!